docs/cookbook/read-and-write-files.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Read and write files
+sidebar_label: Read and write files
+---
+
+In some cases, you need to read and write files to disk. 
+For example, you might need to persist data across app launches, or download data from the internet and save it for later offline use.
+
+Flet makes it easy to work with files and directories on the mobile/desktop device, as seen in the following example.
+
+### Storage Paths
+
+Flet provides two directory paths for data storage, available as environment variables: [`FLET_APP_STORAGE_DATA`](/docs/reference/environment-variables/#flet_app_storage_data) and [`FLET_APP_STORAGE_TEMP`](/docs/reference/environment-variables/#flet_app_storage_temp).
+
+Their values can be gotten as follows:
+
+```python
+import os
+
+app_data_path = os.getenv("FLET_APP_STORAGE_DATA")
+app_temp_path = os.getenv("FLET_APP_STORAGE_TEMP")
+```
+
+### Writing to a File
+
+To write data to a new/existing file, you can use the built-in [`open`](https://docs.python.org/3/library/functions.html#open) function. 
+
+For example: 
+
+```python
+import os
+
+app_data_path = os.getenv("FLET_APP_STORAGE_DATA")
+my_file_path = os.path.join(app_data_path, "test_file.txt")
+
+with open(counter_file_path, "w") as f:
+    f.write("Some file content...")
+```
+
+### Reading from a File
+
+To read data from an existing file, you can equally use the built-in [`open`](https://docs.python.org/3/library/functions.html#open) function.
+
+For example:
+
+```python
+import os
+
+app_app_data_path = os.getenv("FLET_APP_STORAGE_DATA")
+my_file_path = os.path.join(app_app_data_path, "test_file.txt")
+
+with open(counter_file_path, "r") as f:
+    file_content = f.read()
+    print(file_content)
+```
+
+Also, you can use the [`os`](https://docs.python.org/3/library/os.html) module (or any other out there) to perform various file operations like renaming, deleting, listing files present in the directory, etc.
+
+## Example: Counter App
+
+Below is an example that showcases a basic Counter application, whose value persists across app launches. 
+This is made possible by writing the counter value to a file in the app's data storage directory and reading it when the app launches.
+
+<img src="/img/docs/cookbook/read-and-write-files/counter.png" className="screenshot-50" />
+
+```python
+import os
+from datetime import datetime
+import flet as ft
+
+# constants
+FLET_APP_STORAGE_DATA = os.getenv("FLET_APP_STORAGE_DATA")
+COUNTER_FILE_PATH = os.path.join(FLET_APP_STORAGE_DATA, "counter.txt")
+FLET_APP_CONSOLE = os.getenv("FLET_APP_CONSOLE")
+
```

Comment:
Also, the question "what if FLET_APP_STORAGE_DATA is not available"?
Do we need some sort of fallback or throw an exception?

Reply:
As discussed, i'll implement `page.storage` methods for better access to these paths, and in it we can throw exceptions when a path is not available. Then will update this page.

I think we can leave as-is for now?